libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -436,21 +450,38 @@ public void cleanResyncGhosts(long syncId) throws JSONException, IOException {
         final SyncDownTarget target = (SyncDownTarget) sync.getTarget();
 
         // Ask target to clean up ghosts
-        final int localIdSize = target.cleanGhosts(this, soupName, syncId);
```

Comment:
cleanGhosts was not being executed by the thread pool.
(I need to check the native smartsync tests - they are bound to fail now)

Reply:
Ah - failed tests. Maybe you can figure out a way to prevent them from flapping too :-)